diff --git a/m4/np_mysqlclient.m4 b/m4/np_mysqlclient.m4
index c2a4d2a7..5099a02b 100644
--- a/m4/np_mysqlclient.m4
+++ b/m4/np_mysqlclient.m4
@@ -53,18 +53,34 @@ AC_DEFUN([np_mysqlclient],
       _savedcppflags="$CPPFLAGS"
       CPPFLAGS="$CPPFLAGS $np_mysql_include"
 
-      dnl Putting $np_mysql_libs as other libraries ensures that all mysql dependencies are linked in
-      dnl Although -lmysqlclient is duplicated, it is not a problem
-      AC_CHECK_LIB([mysqlclient], [mysql_init], [
-        with_mysql=$np_mysql_config
-        AC_DEFINE(HAVE_MYSQLCLIENT, 1, [Defined if mysqlclient is found and can compile]) 
-	], [with_mysql=no], [$np_mysql_libs])
+      np_check_lib_mysqlclient
+
       CPPFLAGS=$_savedcppflags
 
     fi
   fi
 ])
 
+dnl Test mysql_init using mysqlclient
+AC_DEFUN([np_check_lib_mysqlclient],
+[
+  dnl Putting $np_mysql_libs as other libraries ensures that all mysql dependencies are linked in
+  dnl Although -lmysqlclient is duplicated, it is not a problem
+  AC_CHECK_LIB([mysqlclient], [mysql_init], [
+    with_mysql=$np_mysql_config
+    AC_DEFINE(HAVE_MYSQLCLIENT, 1, [Defined if mysqlclient is found and can compile])
+    ], [np_check_lib_mariadbclient], [$np_mysql_libs])
+])
+
+dnl Test mysql_init using mariadbclient
+AC_DEFUN([np_check_lib_mariadbclient],
+[
+  AC_CHECK_LIB([mariadbclient], [mysql_init], [
+    with_mysql=$np_mysql_config
+    AC_DEFINE(HAVE_MYSQLCLIENT, 1, [Defined if mariadbclient is found and can compile])
+    ], [with_mysql=no], [$np_mysql_libs])
+])
+
 dnl Will take $1, find last occurrance of -LDIR and add DIR to LD_RUN_PATH
 AC_DEFUN([np_add_to_runpath], 
 [
